### PR TITLE
services.openssh: rename several settings

### PR DESCRIPTION
**Pull request by teto**
Follow up of https://github.com/NixOS/nixpkgs/pull/193757

... to match the sshd config format (makes transition smoother), namely:
services.openssh.forwardX11 -> services.openssh.settings.X11Forwarding
services.openssh.cyphers -> services.openssh.settings.Cyphers
services.openssh.macs -> services.openssh.settings.Macs
services.openssh.kexAlgorithms -> services.openssh.settings.KexAlgorithms
services.openssh.gatewayPorts -> services.openssh.settings.GatewayPorts

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by qbit** [comment]
Copying over my comment from https://github.com/NixOS/nixpkgs/pull/193757 for visibility.

Perhaps something like what was done with the git module ( https://github.com/NixOS/nixpkgs/pull/207063 ) could be done here. This would allow arbitrary ordering of options (for Include and what not)

**Comment by teto** [comment]
@qbit looks interesting. I haven't had the need for such a scheme yet but anyone can have a go a it.

@AndersonTorres here is a follow up. I've also noticed that the ssh client used the same options. With my changes they have now diverged which feels strange. How hard do you think it would be to reconcile both, e.g., have ssh and sshd use the same config (at least partially).

File: nixos/doc/manual/release-notes/rl-2305.section.md [hunk 1]
```diff
@@ -124,7 +124,17 @@ In addition to numerous new and upgraded packages, this release has the followin
 
 - The module `usbmuxd` now has the ability to change the package used by the daemon. In case you're experiencing issues with `usbmuxd` you can try an alternative program like `usbmuxd2`. Available as [services.usbmuxd.package](#opt-services.usbmuxd.package)
 
-- A few openssh options have been moved from extraConfig to the new freeform option `settings` and renamed as follow: `services.openssh.kbdInteractiveAuthentication` to `services.openssh.settings.KbdInteractiveAuthentication`, `services.openssh.passwordAuthentication` to `services.openssh.settings.PasswordAuthentication`, `services.openssh.useDns` to `services.openssh.settings.UseDns`, `services.openssh.permitRootLogin` to `services.openssh.settings.PermitRootLogin`, `services.openssh.logLevel` to `services.openssh.settings.LogLevel`.
+- A few openssh options have been moved from extraConfig to the new freeform option `settings` and renamed as follow:
```
Comment (AndersonTorres) [inline nixos/doc/manual/release-notes/rl-2305.section.md:127]:
```suggestion
- A few openssh options have been moved from extraConfig to the new freeform option `settings` and renamed as follows:
```

**Comment by AndersonTorres** [comment]
> How hard do you think it would be to reconcile both, e.g., have ssh and sshd use the same config (at least partially).

There is no reason to seek equality between them. One is a server/daemon, other is a client. They are different programs to different (albeit related) purposes.

I like/prefer the use of structured `settings` mimicking the configfile. And there are similar but different options between ssh and sshd, like `forwardX11` and `x11Forwarding`...

**Comment by teto** [comment]
> There is no reason to seek equality between them. One is a server/daemon, other is a client. They are different programs to different (albeit related) purposes.

except they are both using openssh aka the same program ? Hence most options have similar names. True for forwardX11 and X11Forwarding though ! anyway it's not like I want to do the work, it's just sthg I wanted to mention. This MR is ready from my point of view so far.

File: nixos/doc/manual/release-notes/rl-2305.section.md (hunk 1 above)
Comment (teto) [reply nixos/doc/manual/release-notes/rl-2305.section.md]: dome

**Comment by AndersonTorres** [comment]
About the commits, we don't use [Conventional Commits](conventionalcommits.org) in Nixpkgs.

Our convention is in (github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md)[CONTRIBUTING.md].

**Comment by teto** [comment]
yeah, I just like to keep the generated part separated, makes it easier to edit history IMO. I think squashing is fine here ?

**Comment by AndersonTorres** [comment]
Yes. Squash all!

**Comment by maxeaubrey** [comment]
This breaks initrd ssh.

```
error: attribute 'ciphers' missing

       at /home/systems/nixpkgs/nixos/modules/system/boot/initrd-ssh.nix:132:38:

          131|       KexAlgorithms ${concatStringsSep "," sshdCfg.kexAlgorithms}
          132|       Ciphers ${concatStringsSep "," sshdCfg.ciphers}
             |                                      ^
          133|       MACs ${concatStringsSep "," sshdCfg.macs}
       Did you mean cyphers?
```

Looks like it's because of a typo in the parameter renaming/release notes - cyphers instead of ciphers.

**Comment by misuzu** [comment]
Replacing `services.openssh.cyphers` with `services.openssh.settings.Cyphers` also doesn't fix eval (value is a list of strings):
```
error: A definition for option `services.openssh.settings.Cyphers' is not of type `atom (null, bool, int, float or string)'. Definition values:
       - In `/nix/store/bisgijw38nh20kqkxa1x8f3y8jjyng2s-source/common/openssh.nix':
           [
             "chacha20-poly1305@openssh.com"
           ]
(use '--show-trace' to show detailed location information)
```

**Comment by ElvishJerricco** [comment]
#215179

**Pull request merged**